### PR TITLE
Focus the first attribute field when opening the modal

### LINKED ISSUE #39409: [Product Block Editor] Open the "Add product options" modal
**Description**

If the user clicks "Add options", we'll open the "Add product options" modal, and the entire flow reuses the attributes creation flow: [AkVNGImLgSqCObTQ3idVn7-fi-1637_205314](https://href.li/?https://www.figma.com/file/AkVNGImLgSqCObTQ3idVn7/?node-id=1637-205314)

**Note**

The [API endpoint](https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-product-attributes) does not support pagination and does not filter attributes by `Used for variations`.
The attributes dropdown does not support pagination on scroll.

**ACs**

- [x] Add support for pagination to the attributes field
- [ ] Add support for filtering to the attributes field
- [ ] The option dropdown field immediately focuses on the open attribute field, which lists all custom and local attributes with ‘Used for variations’ enabled. We'd show the first 10 and load the rest when the user starts scrolling inside the dropdown.

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Partially close #39409

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure `New product editor` is turned on under `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`
2. Visit `/wp-admin/admin.php?page=wc-admin&path=/add-product&tab=variations`
3. Then click `Add variation options` button
4. When the modal opened the first attribute field should receive the focus

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/components/attribute-control/new-attribute-modal.tsx [hunk 1]
```diff
@@ -201,6 +206,25 @@ export const NewAttributeModal: React.FC< NewAttributeModalProps > = ( {
...
+
+		if ( ! firstAttributeFieldLabel ) return;
+
+		const timeoutId = setTimeout(
+			( label: HTMLLabelElement ) => {
+				label.focus();
+			},
+			0,
+			firstAttributeFieldLabel
+		);
```
Comment (octaedro) [inline packages/js/product-editor/src/components/attribute-control/new-attribute-modal.tsx]: Is this `setTimeout` necessary? I'm testing on my end, and it worked well without it.
If we decide to keep the `setTimeout` I think we won't need to send the 3rd param `firstAttributeFieldLabel`.

**Review by octaedro — COMMENTED** [review]
Good job @mdperez86! This is testing well on my end I only left a small comment regarding the `setTimeout`.

File: packages/js/product-editor/src/components/attribute-control/new-attribute-modal.tsx (hunk 1 above)
Comment (mdperez86) [reply packages/js/product-editor/src/components/attribute-control/new-attribute-modal.tsx]: done https://github.com/woocommerce/woocommerce/pull/39549/commits/a0190e65e1978daea55dcf1b7106d203df2a20d9

**Review by octaedro — APPROVED** [review]
Thank you for addressing the change. LGTM 🚀

**Pull request merged**